File: app-flutter/lib/pages/flutter/widget/form/showDatePicker.dart
import 'package:flutter_web/material.dart';
// ======================================================================================================== 全局配置
import 'package:flutter_web_study.app/config.dart';
// ======================================================================================================== Widget 控件
class ShowDatePickerPage extends StatefulWidget { // 有状态的控件，控件的状态、文字等属性存在变化的可能
    final String title;
    ShowDatePickerPage({ Key key, this.title }) : super(key: key);
    @override
    _PageState createState() => _PageState(); // === createState()  => _PageState();
}

class _PageState extends State<ShowDatePickerPage> {
    // page
    @override
    Widget build(BuildContext context) {
        return Scaffold(
            appBar: Config.appBar(widget.title),
            body: ListView(
                padding:EdgeInsets.all(10.0),
                children: <Widget>[
                    MaterialButton(
                        color: Colors.red,
                        child: Text('弹出时间选择器', style: TextStyle(color:Colors.white),),
                        onPressed: (){
                            showDatePicker(
                                context: context,
                                initialDate: DateTime.now(),
                                firstDate: DateTime(2018),
                                lastDate: DateTime(2030),
                                builder: (BuildContext context, Widget child) {
                                    return Theme(
                                        data: ThemeData.dark(),
                                        child: child,
                                    );
                                },
                                selectableDayPredicate:(DateTime time){
//                                    print(time);
                                    return true;
                                },
                            ).then((DateTime val) {
                                print(val);   // 2018-07-12 00:00:00.000
                            }).catchError((err) {
                                print(err);
                            });
                        },
                    ),
                    MaterialButton(
                        color: Colors.blue,
                        child: Text('弹出时间选择器', style: TextStyle(color:Colors.white),),
                        onPressed: (){
                            showDatePicker(
                                context: context,
                                initialDate: DateTime.now(),
                                firstDate: DateTime(2018),
                                lastDate: DateTime(2030),
                                builder: (BuildContext context, Widget child) {
                                    return Theme(
                                        data: ThemeData.light(),
                                        child: child,
                                    );
                                },
                                initialDatePickerMode: DatePickerMode.year,
                            ).then((DateTime val) {
                                print(val);   // 2018-07-12 00:00:00.000
                            }).catchError((err) {
                                print(err);
                            });
                        },
                    )
                ],
            )
        );
    }
}

File: app-react/src/pages/flutter/widget/form/showDatePicker.js
import React from 'react'
// ===================================================================== public component
import ListComponent from '@cpt/list.component'
// ===================================================================== component
const data = {
	title:'日期选择器',
	keyword:'showDatePicker( )',
	type:'showDatePicker',
	url:'ShowDatePickerPage',
	data:[
		{ keyword:'initialDate', type:'DateTime', pro:'当前默认时间' },
		{ keyword:'firstDate', type:'DateTime', pro:'开始时间' },
		{ keyword:'lastDate', type:'DateTime', pro:'结束时间' },
		{ keyword:'context', type:'BuildContext', pro:'' },
		
		{ keyword:'initialDatePickerMode', type:'DatePickerMode', pro:'选择器类型', code:[{path:'flutter/widget/Form/showDatePicker/DatePickerMode'}] },
		{ 
			keyword:'builder', type:'Widget Function(BuildContext, Widget)', pro:'配置主题', 
			code:[ {path:'flutter/widget/Form/showDatePicker/builder'} ],
			link:[{ title:'Theme', url:'/flutter/app/Theme' }]
		},
		
		{ keyword:'locale', type:'Locale', pro:'语言', link:'/flutter/style/locale/Locale' },
		{ keyword:'textDirection', type:'TextDirection', pro:'排列顺序', code:[ {path:'flutter/widget/text/TextDirection'} ] },
		
		{ keyword:'selectableDayPredicate', type:'bool Function( DateTime )', pro:'可选择的时间', code:[{path:'flutter/widget/Form/showDatePicker/selectableDayPredicate'}] },
		{ 
			noOrder:true, pro:'示例', 
			code:[ {path:'flutter/widget/Form/showDatePicker/showDatePicker'} ],
			link:[
				{ title:'MaterialButton', url:'/flutter/button/MaterialButton' },
				{ title:'Theme', url:'/flutter/app/Theme' },
				{ title:'DateTime', url:'/dart/time/DateTime' },
			]
		},
	]
}
export default class Index extends React.Component{ render(){ return <ListComponent data={data} {...this.props}/> } }